### PR TITLE
Balance

### PR DESCRIPTION
**Pull request by l34Um1**
Huge patch, all of the things I wanted to do for a while. Every change has a description.

### CODE REVIEW

**Comment by l34Um1** [comment]
I did some things that might need Lua:

enigma midnight pulse to magic damage
make pango e dispellable
make lich e affect ancients

File: game/scripts/npc/abilities/ancient_apparition_cold_feet.txt [hunk 1]
```diff
@@ -59,7 +59,7 @@
       "04"
       {
         "var_type"                                        "FIELD_INTEGER"
-        "cast_range_tooltip"                              "700 800 900 1000 1300 1600"
+        "cast_range_tooltip"                              "700 800 900 1000 1000 1000"
```
Comment (imaGecko) [inline game/scripts/npc/abilities/ancient_apparition_cold_feet.txt]: needs to match "AbilityCastRange"

File: game/scripts/npc/abilities/enigma_midnight_pulse.txt [hunk 2]
```diff
@@ -9,7 +9,7 @@
     //-------------------------------------------------------------------------------------------------------------
     "ID"                                                  "5148"                            // unique ID number for this ability.  Do not change this once established or it will invalidate collected stats.
     "AbilityBehavior"                                     "DOTA_ABILITY_BEHAVIOR_AOE | DOTA_ABILITY_BEHAVIOR_POINT | DOTA_ABILITY_BEHAVIOR_IGNORE_BACKSWING"
-    "AbilityUnitDamageType"                               "DAMAGE_TYPE_PURE"
+    "AbilityUnitDamageType"                               "DAMAGE_TYPE_MAGICAL"
```
Comment (imaGecko) [inline game/scripts/npc/abilities/enigma_midnight_pulse.txt]: should rather open an issue than putting in incorrect tooltips

File: game/scripts/npc/abilities/ancient_apparition_cold_feet.txt (hunk 1 above)
Comment (chrisinajar) [reply game/scripts/npc/abilities/ancient_apparition_cold_feet.txt]: this @l34Um1

**Pull request merged**